### PR TITLE
mgr/volumes: cleanup on fs create error

### PR DESCRIPTION
**Pull request by joscollin**
Fixes:
```
$ ./bin/ceph osd lspools
1 cephfs.a.meta
2 cephfs.a.data
$ ./bin/ceph fs volume create jcollin
Error EINVAL: Creation of multiple filesystems is disabled.  To enable this experimental feature, use 'ceph fs flag set enable_multiple true'
[jcollin@stratocaster build]$ ./bin/ceph fs volume create jcollin1
Error EINVAL: Creation of multiple filesystems is disabled.  To enable this experimental feature, use 'ceph fs flag set enable_multiple true'
$ ./bin/ceph fs volume create jcollin12
Error EINVAL: Creation of multiple filesystems is disabled.  To enable this experimental feature, use 'ceph fs flag set enable_multiple true'
$ ./bin/ceph fs volume create jcollin123
Error EINVAL: Creation of multiple filesystems is disabled.  To enable this experimental feature, use 'ceph fs flag set enable_multiple true'
$ ./bin/ceph osd lspools
1 cephfs.a.meta
2 cephfs.a.data
3 cephfs.jcollin.meta
4 cephfs.jcollin.data
5 cephfs.jcollin1.meta
6 cephfs.jcollin1.data
7 cephfs.jcollin12.meta
8 cephfs.jcollin12.data
9 cephfs.jcollin123.meta
10 cephfs.jcollin123.data
```

### CODE REVIEW

**Comment by joscollin** [comment]
retest this please

File: src/pybind/mgr/volumes/fs/volume.py [hunk 1]
```diff
@@ -309,6 +309,9 @@ def create_volume(self, volname):
         r, outb, outs = self.create_filesystem(volname, metadata_pool, data_pool)
         if r != 0:
             log.error("Filesystem creation error: {0} {1} {2}".format(r, outb, outs))
+            #cleanup the pools
+            self.remove_pool(data_pool)
+            self.remove_pool(metadata_pool)
```
Comment (batrick) [inline src/pybind/mgr/volumes/fs/volume.py]: Also clean up metadata pool if creating the data pool results in an error.

**Comment by joscollin** [comment]
jenkins test make check

**Comment by joscollin** [comment]
retest this please

**Comment by ajarr** [comment]
@joscollin next time create a tracker ticket, else we'll surely miss backporting this.

**Comment by joscollin** [comment]
> @joscollin next time create a tracker ticket, else we'll surely miss backporting this.

Nothing breaks here. It's a clean-up. So no need to backport this.

**Pull request merged**